storage/google/cloud/storage/bucket.py

```diff
@@ -748,22 +751,30 @@ def get_blob(
         :rtype: :class:`google.cloud.storage.blob.Blob` or None
         :returns: The blob object if it exists, otherwise None.
         """
+        import time
+        import random
+
         blob = Blob(
             bucket=self,
             name=blob_name,
             encryption_key=encryption_key,
             generation=generation,
             **kwargs
         )
-        try:
-            # NOTE: This will not fail immediately in a batch. However, when
-            #       Batch.finish() is called, the resulting `NotFound` will be
-            #       raised.
-            blob.reload(client=client)
-        except NotFound:
-            return None
-        else:
-            return blob
+        for retry in range(self.RETRIES):  # pragma: no branch
```

Comment:
Start from 1 and remove `-1` from `self.RETRIES - 1`.

`range(1, self.RETRIES)`

Reply:
If we start range from 1, then there would be 1 less retry. In this case it would be 3 instead of 4 retries.

`if retry == self.RETRIES - 1:`
This is added to check the last retry, where we need to actually raise the exception instead of waiting.

But yes, I have removed "retry - 1" from below code as it was actually adding only 0.5 secs instead of 1, since 2**(0-1) was 0.5. Good catch.
`time.sleep(min(rand_num + 2 ** (retry - 1), rand_num + 32))`
replaced it with
`time.sleep(min(rand_num + 2 ** (retry), rand_num + 32))`